### PR TITLE
CMS 536 breaking news banner

### PR DESCRIPTION
**Pull request by walsh9**
Implement the breaking news banner.

- [x] needs to merge and use https://github.com/nypublicradio/nypr-design-system/pull/31
- [x] needs to merge and use https://github.com/nypublicradio/nypr-design-system/pull/32
- [x] need to update the api to return something we can recognize as an internal url. Either a relative link or a full path including the request domain.
- uses a helper called `get-route-params` to decide when to use routes (internal) vs urls (external). See the readme for this repo for more information on our custom link-to component.
- ~adds the default mirage scenario to a lot of acceptance tests~
- removes the default mirage scenario in favor of letting the app ignore 404s on secondary apis
- a lot of this is modeled off the work that was done for product banner

### CODE REVIEW

File: app/templates/index.hbs [hunk 1]
```diff
@@ -5,6 +5,23 @@
...
+    {{#if (get-route-params news.link)}}
+      <NyprMBlockUrgent
+        @route={{get-route-params news.link}}
+        @title={{news.title}}
+        @timestamp={{timestamp news.startTime}}
+        @description={{news.description}}
+      />
+    {{else}}
+      <NyprMBlockUrgent
+        @url={{news.link}}
```
Comment (mike-hearn) [inline app/templates/index.hbs]: yeah I see what you were talking about - this isn't ideal to have the same block essentially duplicated w/ only url vs route different between them, but I guess this is the best solution if you don't want to refactor the design system components

side question, what happens if you were to pass in both `@route={{get-route-params news.link}}` AND `@url`, regardless of whether `@route` was undefined - I noticed that the underlying `o.block` component takes both simultaneously, so just curious

Comment (walsh9) [reply app/templates/index.hbs]: Yeah we can put this condition at the top (the front?) or at the bottom in nypr-design-system but i think doing it at the bottom requires updating every design system component that uses links to use, maybe a new smart link component, or maybe something else, and we also have more options there which complicates things again.

File: app/serializers/sitewide-components.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import ApplicationSerializer from './application';
+import DS from 'ember-data';
+import { blockToJSONAPI } from '../utils/wagtail-api';
+
+ export default ApplicationSerializer.extend(DS.EmbeddedRecordsMixin, {
+  attrs: {
+    breakingNews: { embedded: 'always' }
+  },
+  modelNameFromPayloadKey: () => 'sitewide-components',
+  normalizeFindRecordResponse(store, primaryModelClass, payload, id) {
```
[the remark this answers is not in the document]
Comment (walsh9) [reply app/serializers/sitewide-components.js]: makes sense but i don't see this in the other serializers so I'm just going to leave this alone for now.

**Comment by walsh9** [comment]
Made a few updates to this since the last review:
https://github.com/nypublicradio/gothamist-web-client/compare/1e6d53b...7f4cefe

File: app/serializers/sitewide-components.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    payload.breaking_news = payload.breaking_news || [];
+    return {
+      data: {
+        id,
+        type: 'sitewide-components',
+        relationships: {
+          breakingNews: {
+            data: payload.breaking_news
+            .filter(blockIsNotNull)
```
Comment (mike-hearn) [inline app/serializers/sitewide-components.js]: nice

**Pull request merged**